### PR TITLE
Feature/sn 603 fix image display

### PR DESCRIPTION
**Pull request by AlexanderMakarov124**
In this PR I refactored `UploadImage` control, by replacing `MvvmMessenger` with built-in tools.
Also I added `IUploadFileStrategy` interface with implementations:
- Base 64 file
- File in S3 storage (added S3 service to implement this)
- File in file system (it was not used, only for example)

### CODE REVIEW

File: demo/Saritasa.NetForge.Demo/Infrastructure/UploadFiles/Strategies/UploadFileToFileSystemStrategy.cs [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+        await file.OpenReadStream(cancellationToken: cancellationToken).CopyToAsync(fileStream, cancellationToken);
+
+        return path;
+    }
+
+    /// <summary>
+    /// Returns path to file as <paramref name="fileString"/> as it is.
+    /// </summary>
+    /// <param name="fileString"></param>
+    /// <returns></returns>
```
Comment (toanphamsaritasa) [inline demo/Saritasa.NetForge.Demo/Infrastructure/UploadFiles/Strategies/UploadFileToFileSystemStrategy.cs:33]: Empty return.

**Review by toanphamsaritasa — APPROVED** [review]

**Review by mchernikov80 — APPROVED** [review]

File: demo/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,32 @@
+FROM mcr.microsoft.com/dotnet/sdk:8.0 AS publish
+
+COPY ./src src
+COPY ./GitVersion.yml GitVersion.yml
```
Comment (vasiliy-chefonov) [inline demo/Dockerfile:4]: Do we need this one?

File: demo/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,32 @@
+FROM mcr.microsoft.com/dotnet/sdk:8.0 AS publish
```
Comment (vasiliy-chefonov) [inline demo/Dockerfile:1]: What is the purpose of this docker file?

File: src/Saritasa.NetForge.Domain/IUploadFileStrategy.cs [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+    /// </summary>
+    /// <param name="file">File to upload.</param>
+    /// <param name="cancellationToken">Token to cancel the operation.</param>
+    /// <returns>File related object.</returns>
+    Task<object> UploadFileAsync(IBrowserFile file, CancellationToken cancellationToken);
+
+    /// <summary>
+    /// Gets file source.
+    /// </summary>
+    /// <param name="fileString">File related string.</param>
```
Comment (vasiliy-chefonov) [inline src/Saritasa.NetForge.Domain/IUploadFileStrategy.cs:21]: What is the file-related string?

File: demo/Saritasa.NetForge.Demo/Infrastructure/UploadFiles/S3Storage/ContentUriBuilder.cs [hunk 5]
```diff
@@ -0,0 +1,30 @@
+﻿using MimeTypes;
+
+namespace Saritasa.NetForge.Demo.Infrastructure.UploadFiles.S3Storage;
+
+/// <summary>
+/// Contains methods for creating content URI
```
Comment (vasiliy-chefonov) [inline demo/Saritasa.NetForge.Demo/Infrastructure/UploadFiles/S3Storage/ContentUriBuilder.cs:6]: Make it single line.

File: demo/Saritasa.NetForge.Demo/Infrastructure/UploadFiles/S3Storage/ContentUriBuilder.cs [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+/// for files to upload them.
+/// </summary>
+public static class ContentUriBuilder
```
Comment (vasiliy-chefonov) [inline demo/Saritasa.NetForge.Demo/Infrastructure/UploadFiles/S3Storage/ContentUriBuilder.cs:9]: Can be internal

File: demo/Saritasa.NetForge.Demo/Infrastructure/UploadFiles/S3Storage/S3StorageService.cs [hunk 7]
```diff
@@ -0,0 +1,253 @@
...
+        s3Client = ConfigureAmazonS3Client();
+        bucketName = settings.BucketName;
+    }
+
+    /// <inheritdoc />
+    public async Task UploadAsync(string key, Stream stream, CancellationToken cancellationToken)
+    {
+        try
+        {
+            await CreateBucketIfNotExistsAsync(cancellationToken);
```
Comment (vasiliy-chefonov) [inline demo/Saritasa.NetForge.Demo/Infrastructure/UploadFiles/S3Storage/S3StorageService.cs:43]: Do we really need to create the bucker? What about the bucket permissions?

File: demo/Saritasa.NetForge.Demo/Infrastructure/UploadFiles/Strategies/UploadBase64FileStrategy.cs [hunk 8]
```diff
@@ -0,0 +1,39 @@
+﻿using Microsoft.AspNetCore.Components.Forms;
+using Saritasa.NetForge.Domain;
+
+namespace Saritasa.NetForge.Demo.Infrastructure.UploadFiles.Strategies;
+
+/// <summary>
+/// Strategy when uploaded file converts to base 64 string.
+/// </summary>
+public class UploadBase64FileStrategy : IUploadFileStrategy
```
Comment (vasiliy-chefonov) [inline demo/Saritasa.NetForge.Demo/Infrastructure/UploadFiles/Strategies/UploadBase64FileStrategy.cs:9]: Can be internal

**Review by vasiliy-chefonov — APPROVED** [review]

File: src/Saritasa.NetForge.Domain/IUploadFileStrategy.cs (hunk 4 above)
Comment (AlexanderMakarov124) [reply src/Saritasa.NetForge.Domain/IUploadFileStrategy.cs:21]: Updated comments in `IUploadedFileStrategy` to clarify such abstract moments

File: demo/Saritasa.NetForge.Demo/Infrastructure/UploadFiles/S3Storage/S3StorageService.cs (hunk 7 above)
Comment (AlexanderMakarov124) [reply demo/Saritasa.NetForge.Demo/Infrastructure/UploadFiles/S3Storage/S3StorageService.cs:43]: Yes, bucket is required to upload files to S3

Comment (vasiliy-chefonov) [reply demo/Saritasa.NetForge.Demo/Infrastructure/UploadFiles/S3Storage/S3StorageService.cs:43]: I mean why do we need to create the new one if it doesn't exist?

**Pull request merged**